### ISSUE TITLE
resetNavigationProgress is not working in dev mode with react 18 & strict mode

### ISSUE THREAD

**Issue by matinrco** [mantinedev/mantine#2057]
### What package has an issue

@mantine/nprogress

### Describe the bug

Hey! 👋
Its kinda related to react 18 useEffect double call in dev mode. 🤔
otherwise all things works great with react 17 or disabled strict mode in react 18.
I'm using nextjs (12). `@mantine/nprogress` is configured like the [docs](https://mantine.dev/others/nprogress/#usage-with-nextjs) exactly.
Test scenario:
1- you are in page A.
2 - navigate to page B with `<Link/>`. (progress works correctly)
3- go back from page B to A. `resetNavigationProgress` is called but has no effect on progress bar.

### What version of @mantine/hooks page do you have in package.json?

5.1.1

### If possible, please include a link to a codesandbox with the reproduced problem

[sandbox link](https://codesandbox.io/s/mantine-nprogress-test-7qp1l7)

### Do you know how to fix the issue

No

**Comment by rtivital** [comment]
Does it work fine in production?

**Comment by matinrco** [comment]
> Does it work fine in production?

yes, production build works fine.

**Comment by rtivital** [comment]
Okay, great, I'll be trying to fix the issue with event based function in next patch, hope that you will be able to validate the fix

**Comment by matinrco** [comment]
> Okay, great, I'll be trying to fix the issue with event based function in next patch, hope that you will be able to validate the fix

Its great, thank you!
I will do whatever I can.

p.s: sandbox link added.
go to page A.
then, click on "go to page B".
after visiting page B, hit back button.
progress never ends.

sandbox is based on official mantine nextjs template.

**Comment by rtivital** [comment]
Fixed in 5.1.3 @matinrco can you please verify that the fix works for you?

**Comment by matinrco** [comment]
> Fixed in 5.1.3 @matinrco can you please verify that the fix works for you?

Unfortunately not :/

I did a little inspection. might be helpful:

in mantine 5.1.3:
progress bar doesn't even start. `fnRef.current` value in `start` function which is exposed from `useInterval` is `undefined`. but idk is this a root cause or not.

in mantine 5.1.1:
`start` function which is exposed from `useInterval` is called twice! which means that if we call `interval.stop()` in `NavigationProgress`, the last interval will be cleared and the previous one remains headless which means progress bar never stops in UI.
so what I did was to add...
```ts
window.clearInterval(intervalRef.current);
intervalRef.current = null;
```
before setting new interval. before [this line](https://github.com/mantinedev/mantine/blob/813aaac49200bc5306793777dfbf60bb0ac654a0/src/mantine-hooks/src/use-interval/use-interval.ts#L15).
and it worked! but idk is this a correct workaround or not.
besides, situation is different in 5.1.3 release :/
